sw/device/lib/testing/test_framework/freertos_port.c

```diff
@@ -0,0 +1,103 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_rv_timer.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/vendor/freertos_freertos_kernel/include/FreeRTOS.h"
+#include "sw/vendor/freertos_freertos_kernel/include/task.h"
+#include "sw/vendor/freertos_freertos_kernel/portable/GCC/RISC-V/portmacro.h"
+
+// TODO: make this toplevel agnostic.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+// NOTE: some of the function names below do NOT, and cannot, conform to the
+// style guide, since they are specific implementations of FreeRTOS defined
+// functions.
+
+// ----------------------------------------------------------------------------
+// Timer Setup (for use when preemptive scheduling is enabled)
+// ----------------------------------------------------------------------------
+#if defined(configUSE_PREEMPTION) && configUSE_PREEMPTION == 1
```

Comment:
Is `#ifdef <macro>` not enough?

Reply:
After a quick check ([sw/vendor/freertos_freertos_kernel/include/FreeRTOS.h:95](https://github.com/lowRISC/opentitan/blob/master/sw/vendor/freertos_freertos_kernel/include/FreeRTOS.h#L95)) FreeRTOS _requires_ configUSE_PREEMPTION be defined, so I simplified this to "#if configUSE_PREEMPTION == 1"